test/cql-pytest/test_describe.py

```diff
@@ -447,6 +447,75 @@ def test_desc_index(cql, test_keyspace):
         
         assert f"CREATE INDEX named_index ON {tbl}{maybe_space}(c)" in c_desc
 
+def test_desc_index_on_collections(cql, test_keyspace):
+    # In this test, all assertions are in form of
+    # `assert create_stmt in desc or create_stmt.replace("(", " (", 1)`
+    # This is because Scylla and Cassandra have slightly different (whitespace-wise)
+    # output of `DESC INDEX`. 
+    #   Scylla:     `CREATE INDEX ON table_name(column_name)`
+    #   Cassandra:  `CREATE INDEX ON table_name (column_name)`
+    
+    with new_test_table(cql, test_keyspace, "a int primary key, b int, c frozen<set<int>>, d map<int, int>, e list<int>, f set<int>, g list<int>, h set<int>") as tbl:
+        indexes = [
+            ("idx_b",           f"CREATE INDEX idx_b ON {tbl}(b)"),
+            ("idx_c",           f"CREATE INDEX idx_c ON {tbl}(full(c))"),
+            ("idx_d_entries",   f"CREATE INDEX idx_d_entries ON {tbl}(entries(d))"),
+            ("idx_d_keys",      f"CREATE INDEX idx_d_keys ON {tbl}(keys(d))"),
+            ("idx_d_values",    f"CREATE INDEX idx_d_values ON {tbl}(values(d))"),
+            ("idx_e",           f"CREATE INDEX idx_e ON {tbl}(values(e))"),
+            ("idx_f",           f"CREATE INDEX idx_f ON {tbl}(values(f))"),
+        ]
+
+        for name, create_stmt in indexes:
+            cql.execute(create_stmt)
+            desc = cql.execute(f"DESC INDEX {test_keyspace}.{name}").one().create_statement
+            assert create_stmt in desc or create_stmt.replace("(", " (", 1)
+        
+        # When an index is created on list column with `CREATE INDEX ON tbl(list_column)`,
+        # its target is `values(list_column)` and describe will print creat statement with that
+        # values() function added.
+        cql.execute(f"CREATE INDEX idx_g ON {tbl}(g)")
+        idx_g_desc = cql.execute(f"DESC INDEX {test_keyspace}.idx_g").one().create_statement
+        idx_g_create_stmt = f"CREATE INDEX idx_g ON {tbl}(values(g))"
+        assert idx_g_create_stmt in idx_g_desc or idx_g_create_stmt.replace("(", " (", 1)
+
+        # The same situation but with set
+        cql.execute(f"CREATE INDEX idx_h ON {tbl}(h)")
+        idx_h_desc = cql.execute(f"DESC INDEX {test_keyspace}.idx_h").one().create_statement
+        idx_h_create_stmt = f"CREATE INDEX idx_h ON {tbl}(values(h))"
+        assert idx_h_create_stmt in idx_h_desc or idx_h_create_stmt.replace("(", " (", 1)
+
+def test_desc_index_column_quoting(cql, test_keyspace):
+    # In this test, all assertions are in form of
+    # `assert create_stmt in desc or create_stmt.replace("(", " (", 1)`
+    # This is because Scylla and Cassandra have slightly different (whitespace-wise)
+    # output of `DESC INDEX`. 
+    #   Scylla:     `CREATE INDEX ON table_name(column_name)`
+    #   Cassandra:  `CREATE INDEX ON table_name (column_name)`
+
+    quoted_names = ['"select"', '"hEllo"', '"x y"', '"hi""hello""yo"', '"""hi"""', '"keys(m)"', '"values(m)"', '"entries(m)"']
```

Comment:
I see you did check the case of a column named `keys(m)` but this column is not a list (it's an int) so it doesn't catch the bug I suspected in the previous patch. So I guess if this is a real bug you'll need a separate test for it.

Reply:
changed the test so columns with names keys/values/entries have appropriate types